controllers/topology/cluster_controller.go

```diff
@@ -108,6 +110,7 @@ func (r *ClusterReconciler) Reconcile(ctx context.Context, req ctrl.Request) (_
 		// Error reading the object - requeue the request.
 		return ctrl.Result{}, err
 	}
+	cluster.Kind = "Cluster"
```

Comment:
Why is this required?

Reply:
I think the idea is to set the Kind so the event recorder can set it correctly when creating the event.

But if I saw correctly the event recorder will try to get the kind from the scheme as a fallback:
https://github.com/kubernetes/client-go/blob/master/tools/reference/ref.go#L59-L72

But I might have looked at the wrong code as I just was clicking through endless interfaces, and didn't try it with the debugger or something.

Reply:
The problem is our KObj object reporting a empty kind for the Cluster (it does not have fall back to schema)